### PR TITLE
Provide keyringName configuration to OIDC CredentialsProvider lookup 

### LINKED ISSUE #41265: Quarkus OIDC CredenitalsProvider does not follow conventions of allowing a separate "bean name" and "keyring name"
### Describe the bug

Other `CredentialsProvdider` integrations, such as the ones built by Quarkus Data source, RabbitMQ, and Quarkiverse GitHub app extensions allow for separate configuration of a "bean name" of the `CredentialsProvider` implementation to be used by the extension and a "keyring name". This is also laid out in `CredentialsProvider` documentation [here](https://quarkus.io/guides/credentials-provider#new-credentials-provider-extension). 

However, the Quarkus OIDC extension only allows for the configuration of `quarkus.oidc.credentials.client-secret.provider.name` (which is used [here](https://github.com/quarkusio/quarkus/blob/9e64ec21bb19894e949f4d468b51ad6766974341/extensions/oidc-common/runtime/src/main/java/io/quarkus/oidc/common/runtime/OidcCommonUtils.java#L314) to first lookup the CDI bean and then used [here](https://github.com/quarkusio/quarkus/blob/9e64ec21bb19894e949f4d468b51ad6766974341/extensions/oidc-common/runtime/src/main/java/io/quarkus/oidc/common/runtime/OidcCommonUtils.java#L317) to look up the secret from the keyring 

I think the only way this could be successfully used today is by always supplying what's supposed to be an optional bean name for the `CredentialsProvider` implementation because otherwise the call to `CredentialsProvider.getCredentials(String)` will always be provided `null`.

This mandates that the bean name and the value in the keyring be the same. For my use case, which is a `CredentialsProvider` implementation backed by data in AWS SecretsManager, in means I need a separate `CredentialsProvider` implementation for just this extension where the name of the secret in SecretsManager matches the bean name of the `CredentialsProvider` implementation.

### Expected behavior

We should adapt the OIDC extension to look more like other usages of `CredentialsProviderFinder.find`, like this

```java

 private static Supplier<? extends String> fromCredentialsProvider(Provider provider) {
        return new Supplier<String>() {

            @Override
            public String get() {
                if (provider.key.isPresent()) {
                    String providerName = provider.name.orElse(null);
                    CredentialsProvider credentialsProvider = CredentialsProviderFinder.find(providerName);
                    if (provider.keyringName.isPresent()) {
                        credentialsProvider.getCredentials(provider.keyringName.get()).get(provider.key.get());
                    }
                    else {
                        return credentialsProvider.getCredentials(providerName).get(provider.key.get());
                    }
                }
                return null;
            }
        };
    }
```

This does the following:
1. remove a redundant null check, because `CredentialsProviderFinder.find` throws a `RuntimeException` if the `CredentialsProvider` resolved from the `Arc` container is null, and therefore this method will never return null
2. Adds a new config property for the keyring that is used if it is present.

In a future release of Quarkus, the breaking change to require the user to provide a keyring name (like all other extensions that support `CredentialsProvider` should be considered, but this change was structured in this way to be backward compatible. 

I'm definitely open to other suggestions for implementing a change like this! I'm also appy to contribute this change myself if the workaround to conform to conventions is accepted.

Note, normally `quarkus.oidc.credentials.client-secret.provider.name` would be the name of the keyring, not the name of the CDI bean, but since it's already being used here for both in order to be backward compatible we have to introduce a new property for when users want to configure the keyring name (which is usually non-optional when configuring extensions to use `CredentialsProvider` implementations 

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by ryandens**
Fixes #41265

- Removes redundant null-check on value returned by `CredentialsProviderFinder.find(String)`
- Modifies test to verify the appropriate keyring is provided to the `CredentialsProvider` test implementation
  - Took a slightly different strategy here than suggested in #41265, as concatenating the keyring name to the key would mean that the `Map.get(String)` call on [this line](https://github.com/quarkusio/quarkus/blob/d070b8128e8ebb4bb4330a6d5fe199bede9f73c6/extensions/oidc-common/runtime/src/main/java/io/quarkus/oidc/common/runtime/OidcCommonUtils.java#L324) would return `null` because the extension expects the map key to be the configured value for `quarkus.oidc.credentials.client-secret.provider.key`. Instead, we verify this functionality by only returning the stubbed secret in the `Map` if the keyring name matches our expectations. I think this is desirable as it more maps to how a production-like `CredentialsProvider` implementation would work.
 - Modifies the `OidcCommonUtils` logic to use the configured `keyringName`, or `null` to look up the OIDC client secret from the `CredentialsProvider`, as suggested by @sberyozkin in #41265. I wasn't sure if we should instead access this optional value by wrapping the codeblock in a `Optional.isPresent` check, like we do for the provider key config item, but happy to follow the suggestions of the Quarkus team.
- Updates documentation to reflect the new config item required to use the OIDC extension with a `CredentialsProvider`

Let me know if there's anything else I can do!

### CODE REVIEW

**Comment by sberyozkin** [comment]
Thanks @ryandens, LGTM, can you please rebase to pick up the latest updates, like a Maven version bump to 3.9.8  ?
Please also squash, use the PR title if you don't mind as the single commit message
Thanks

File: extensions/oidc-common/runtime/src/main/java/io/quarkus/oidc/common/runtime/OidcCommonConfig.java [hunk 1]
```diff
@@ -467,12 +467,19 @@ public void setAssertion(boolean assertion) {
...
             /**
-             * The CredentialsProvider name, which should only be set if more than one CredentialsProvider is
+             * The CredentialsProvider bean name, which should only be set if more than one CredentialsProvider is
              * registered
              */
             @ConfigItem
             public Optional<String> name = Optional.empty();
 
+            /**
+             * The CredentialsProvider keyring name, which should always be set when using a CredentialsProvider
```
Comment (sberyozkin) [inline extensions/oidc-common/runtime/src/main/java/io/quarkus/oidc/common/runtime/OidcCommonConfig.java:477]: @ryandens It should remain optional, if my provider just reads a secret from the local database, there should be no need to set this property. Also a missing `n` at the end of the doc

File: docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc [hunk 2]
```diff
@@ -133,6 +133,8 @@ quarkus.oidc.client-id=quarkus-app
 
 # This is a key which will be used to retrieve a secret from the map of credentials returned from CredentialsProvider
 quarkus.oidc.credentials.client-secret.provider.key=mysecret-key
+# this is the keyring provided to the CredentialsProvider when looking up the secret
+quarkus.oidc.credentials.client-secret.provider.keyringName=oidc
```
Comment (sberyozkin) [inline docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:137]:
```suggestion
quarkus.oidc.credentials.client-secret.provider.keyring-name=oidc
```

File: docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc [hunk 3]
```diff
@@ -165,6 +167,8 @@ quarkus.oidc.client-id=quarkus-app
 
 # This is a key which will be used to retrieve a secret from the map of credentials returned from CredentialsProvider
 quarkus.oidc.credentials.jwt.secret-provider.key=mysecret-key
+# this is the keyring provided to the CredentialsProvider when looking up the secret
+quarkus.oidc.credentials.client-secret.provider.keyringName=oidc
```
Comment (sberyozkin) [inline docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:171]:
```suggestion
quarkus.oidc.credentials.client-secret.provider.keyring-name=oidc
```

File: docs/src/main/asciidoc/security-openid-connect-client-reference.adoc [hunk 4]
```diff
@@ -725,6 +725,8 @@ quarkus.oidc-client.client-id=quarkus-app
 
 # This key is used to retrieve a secret from the map of credentials returned from CredentialsProvider
 quarkus.oidc-client.credentials.client-secret.provider.key=mysecret-key
+# this is the keyring provided to the CredentialsProvider when looking up the secret
+quarkus.oidc.credentials.client-secret.provider.keyringName=oidc
```
Comment (sberyozkin) [inline docs/src/main/asciidoc/security-openid-connect-client-reference.adoc:729]:
```suggestion
quarkus.oidc.credentials.client-secret.provider.keyring-name=oidc
```

File: extensions/oidc-client/deployment/src/test/resources/application-oidc-client-credentials.properties [hunk 5]
```diff
@@ -5,4 +5,5 @@ quarkus.oidc.credentials.secret=secret
 quarkus.oidc-client.auth-server-url=${quarkus.oidc.auth-server-url}
 quarkus.oidc-client.client-id=${quarkus.oidc.client-id}
 quarkus.oidc-client.credentials.client-secret.provider.name=vault-secret-provider
+quarkus.oidc-client.credentials.client-secret.provider.keyringName=oidc
```
Comment (sberyozkin) [inline extensions/oidc-client/deployment/src/test/resources/application-oidc-client-credentials.properties:8]:
```suggestion
quarkus.oidc-client.credentials.client-secret.provider.keyring-name=oidc
```

File: docs/src/main/asciidoc/security-openid-connect-client-reference.adoc [hunk 6]
```diff
@@ -757,6 +759,8 @@ quarkus.oidc-client.client-id=quarkus-app
 
 # This is a key that will be used to retrieve a secret from the map of credentials returned from CredentialsProvider
 quarkus.oidc-client.credentials.jwt.secret-provider.key=mysecret-key
+# this is the keyring provided to the CredentialsProvider when looking up the secret
+quarkus.oidc.credentials.client-secret.provider.keyringName=oidc
```
Comment (sberyozkin) [inline docs/src/main/asciidoc/security-openid-connect-client-reference.adoc:763]:
```suggestion
quarkus.oidc.credentials.client-secret.provider.keyring-name=oidc
```

File: extensions/oidc-common/runtime/src/main/java/io/quarkus/oidc/common/runtime/OidcCommonConfig.java (hunk 1 above)
Comment (sberyozkin) [reply extensions/oidc-common/runtime/src/main/java/io/quarkus/oidc/common/runtime/OidcCommonConfig.java:477]: Please update the message to explain when it must be set but let users know they are not required to set it

File: docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc [hunk 7]
```diff
@@ -133,6 +133,8 @@ quarkus.oidc.client-id=quarkus-app
 
 # This is a key which will be used to retrieve a secret from the map of credentials returned from CredentialsProvider
 quarkus.oidc.credentials.client-secret.provider.key=mysecret-key
+# this is the keyring provided to the CredentialsProvider when looking up the secret
```
Comment (sberyozkin) [inline docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:136]: Please tune the message to explain when setting this property is required but let users know when they can omit it

File: extensions/oidc/deployment/src/test/resources/application-dev-mode.properties [hunk 8]
```diff
@@ -2,6 +2,7 @@ quarkus.oidc.auth-server-url=${keycloak.url}/realms/quarkus
 quarkus.oidc.tenant-enabled=false
 quarkus.oidc.client-id=${oidc.client-id}
 quarkus.oidc.credentials.client-secret.provider.name=vault-secret-provider
+quarkus.oidc.credentials.client-secret.provider.keyringName=oidc
```
Comment (ryandens) [inline extensions/oidc/deployment/src/test/resources/application-dev-mode.properties:5]:
```suggestion
quarkus.oidc.credentials.client-secret.provider.keyring-name=oidc
```

**Comment by ryandens** [comment]
Thank you @sberyozkin! I believe I incorporated all your suggestions, I appreciate the help!

File: extensions/oidc-common/runtime/src/main/java/io/quarkus/oidc/common/runtime/OidcCommonConfig.java [hunk 9]
```diff
@@ -467,12 +467,22 @@ public void setAssertion(boolean assertion) {
...
```
Comment (sberyozkin) [inline extensions/oidc-common/runtime/src/main/java/io/quarkus/oidc/common/runtime/OidcCommonConfig.java:477]:
```suggestion
             * The CredentialsProvider keyring name```

File: extensions/oidc-common/runtime/src/main/java/io/quarkus/oidc/common/runtime/OidcCommonConfig.java [hunk 10]
```diff
@@ -467,12 +467,22 @@ public void setAssertion(boolean assertion) {
...
+             * to provide a secret key to this extension. The keyring name is only required when the CredentialsProvider being
```
Comment (sberyozkin) [inline extensions/oidc-common/runtime/src/main/java/io/quarkus/oidc/common/runtime/OidcCommonConfig.java:478]:
```suggestion
             * The keyring name is only required when the CredentialsProvider being
```

File: docs/src/main/asciidoc/security-openid-connect-client-reference.adoc [hunk 11]
```diff
@@ -725,6 +725,8 @@ quarkus.oidc-client.client-id=quarkus-app
 
 # This key is used to retrieve a secret from the map of credentials returned from CredentialsProvider
 quarkus.oidc-client.credentials.client-secret.provider.key=mysecret-key
+# this is the keyring provided to the CredentialsProvider when looking up the secret, set only if required by the CredentialsProvider implementation
```
Comment (sberyozkin) [inline docs/src/main/asciidoc/security-openid-connect-client-reference.adoc:728]:
```suggestion
# This is the keyring provided to the CredentialsProvider when looking up the secret, set only if required by the CredentialsProvider implementation
```

Sorry about that, but I'm pretty sure our Doc or QE team would prefer to align it with capital `T` in the previous comment :-)

File: docs/src/main/asciidoc/security-openid-connect-client-reference.adoc [hunk 12]
```diff
@@ -757,6 +759,8 @@ quarkus.oidc-client.client-id=quarkus-app
 
 # This is a key that will be used to retrieve a secret from the map of credentials returned from CredentialsProvider
 quarkus.oidc-client.credentials.jwt.secret-provider.key=mysecret-key
+# this is the keyring provided to the CredentialsProvider when looking up the secret, set only if required by the CredentialsProvider implementation
```
Comment (sberyozkin) [inline docs/src/main/asciidoc/security-openid-connect-client-reference.adoc:762]:
```suggestion
# This is the keyring provided to the CredentialsProvider when looking up the secret, set only if required by the CredentialsProvider implementation
```

File: docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc [hunk 13]
```diff
@@ -133,6 +133,8 @@ quarkus.oidc.client-id=quarkus-app
 
 # This is a key which will be used to retrieve a secret from the map of credentials returned from CredentialsProvider
 quarkus.oidc.credentials.client-secret.provider.key=mysecret-key
+# this is the keyring provided to the CredentialsProvider when looking up the secret, set only if required by the CredentialsProvider implementation
```
Comment (sberyozkin) [inline docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:136]:
```suggestion
# This is the keyring provided to the CredentialsProvider when looking up the secret, set only if required by the CredentialsProvider implementation
```

File: docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc [hunk 14]
```diff
@@ -165,6 +167,8 @@ quarkus.oidc.client-id=quarkus-app
 
 # This is a key which will be used to retrieve a secret from the map of credentials returned from CredentialsProvider
 quarkus.oidc.credentials.jwt.secret-provider.key=mysecret-key
+# this is the keyring provided to the CredentialsProvider when looking up the secret, set only if required by the CredentialsProvider implementation
```
Comment (sberyozkin) [inline docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:170]:
```suggestion
# This is the keyring provided to the CredentialsProvider when looking up the secret, set only if required by the CredentialsProvider implementation
```

**Comment by sberyozkin** [comment]
Hi @ryandens A few more tiny suggestions are proposed and it will be ready to go, thanks

File: docs/src/main/asciidoc/security-openid-connect-client-reference.adoc (hunk 11 above)
Comment (ryandens) [reply docs/src/main/asciidoc/security-openid-connect-client-reference.adoc:728]: Np! Makes sense to me - I should have noticed the other lines were capitalized

**Review by sberyozkin — APPROVED** [review]
LGTM, we can merge once a few more minor doc updates are applied

**Review by sberyozkin — APPROVED** [review]

**Pull request merged**